File: android/lib/src/main/java/org/elastos/sdk/elephantwallet/contact/internal/EventArgs.java
package org.elastos.sdk.elephantwallet.contact.internal;

public class EventArgs {
    public enum Type {
        StatusChanged(101),
//        ReceivedMessage(102),
//        SentMessage(103),
        FriendRequest(104),
//        FriendStatusChanged(105),
        HumanInfoChanged(106);

        public static Type valueOf(int id) {
            Type[] values = Type.values();
            for(int idx = 0; idx < values.length; idx++) {
                if(values[idx].id == id) {
                    return values[idx];
                }
            }
            return null;
        }

        private Type(int id){
            this.id = id;
        }
        private int id;
    }

    public EventArgs(int type, String humanCode, int channelType, byte[] data) {
        this.type = Type.valueOf(type);
        this.humanCode = humanCode;
        this.channelType = ContactChannel.valueOf(channelType);
        this.data = data;
    }

    @Override
    public String toString() {
        return "EventArgs"
                + "[type=" + type
                + ",humanCode=" + humanCode
                + ",channelType=" + channelType
                + ",data=" + data
                +"]";
    }

    public Type type;
    public String humanCode;
    public ContactChannel channelType;
    public byte[] data;
}

File: ios/sdk/swift/internal/EventArgs.swift

public class EventArgs {
  public enum Kind : Int {
    case StatusChanged = 101
    //        ReceivedMessage = 102
    //        SentMessage = 103
    case FriendRequest = 104
    //        FriendStatusChanged(105),
    case HumanInfoChanged = 106
  }
  
  public init(type: Int, humanCode: String, channelType: Int, data: Data?) {
    self.type = Kind(rawValue: type)!
    self.humanCode = humanCode
    self.channelType = ContactChannel(rawValue: channelType)!
    self.data = data
  }
  
  public func toString() -> String {
    return  "EventArgs"
          + "[type=\(type)"
          + ",humanCode=\(humanCode)"
          + ",channelType=\(channelType)"
          + ",data=\(String(describing: data))"
          + "]"
  }
  
  public let type: Kind
  public let humanCode: String
  public let channelType: ContactChannel
  public let data: Data?
}
